x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptLongMappedFieldType.java

```diff
@@ -51,8 +52,9 @@ public Object valueForDisplay(Object value) {
         return new ScriptLongFieldData.Builder(name(), script, scriptFactory);
     }
 
-    private LongScriptFieldScript.LeafFactory leafFactory(QueryShardContext context) {
-        return scriptFactory.newFactory(script.getParams(), context.lookup());
+    private AbstractLongScriptFieldScript.LeafFactory leafFactory(QueryShardContext context) {
+        LongScriptFieldScript.LeafFactory delegate = scriptFactory.newFactory(script.getParams(), context.lookup());
+        return ctx -> delegate.newInstance(ctx);
```

Comment:
I get confused here on what has changed. why can't we return the long field script. given that we are in the long mapped field type?

Reply:
I don't remember! I'll check.

Reply:
It had to do with being able to share the queries. I've dropped it in favor of something a little more obvious.